server/plugins/public.js

```diff
@@ -95,13 +95,17 @@ function register (server, options, next) {
 
     var is404 = response.output.statusCode === 404
     var isHTML = /text\/html/.test(request.headers.accept)
+    var isResource = /(text\/css|application\/javascript|image\/png|image\/jpeg)/.test(request.headers.accept)
     var isAccountPath = /^\/hoodie\/account\//.test(request.path)
     var isAdminPath = /^\/hoodie\/admin\//.test(request.path)
     var isStorePath = /^\/hoodie\/store\//.test(request.path)
     var isHoodiePath = /^\/hoodie\//.test(request.path)
 
     // We only care about 404 for html requests...
     if (!is404 || !isHTML) {
+      if (is404 && isResource) {
+        return reply(fs.createReadStream(path.join(hoodiePublicPath, '404.html'))).code(404)
+      }
```

Comment:
I think this could be updated in the reverse with:

```diff
-    var isHTML = /text\/html/.test(request.headers.accept)
     var isResource = /(text\/css|application\/json|application\/javascript|image\/png|image\/jpeg)/.test(request.headers.accept) || /.(css|js|png|jpg)$/.test(request.path)
+    var isHTML = !isResource
```